File: business/business-system/src/main/java/com/glowxq/system/meta/enums/TextType.java
package com.glowxq.system.meta.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.system.meta.base.TextEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum TextType implements BaseType, TextEnum {
    Markdown("Markdown", "Markdown"),
    RichText("RichText", "富文本"),
    NormalText("NormalText", "普通文本"),
    ;

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static TextType matchCode(String code) {
        for (TextType textType : TextType.values()) {
            if (textType.getCode().equals(code)) {
                return textType;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/system/meta/TextType.ts
import type { BaseEnum } from '@/enums/base';

// 文本类型枚举接口
interface TextTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 文本类型枚举
 * 用于标识不同的文本类型
 */
export const TextType: { [key: string]: TextTypeEnum } = {
  /**
   * Markdown 文本
   */
  MARKDOWN: {
    code: 'Markdown',
    name: 'Markdown',
    tooltip: 'Markdown格式文本',
    text: 'Markdown格式文本'
  },

  /**
   * 富文本
   */
  RICH_TEXT: {
    code: 'RichText',
    name: '富文本',
    tooltip: '富文本格式',
    text: '富文本格式'
  },

  /**
   * 普通文本
   */
  NORMAL_TEXT: {
    code: 'NormalText',
    name: '普通文本',
    tooltip: '普通文本格式',
    text: '普通文本格式'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): TextTypeEnum | null {
  return Object.values(TextType).find(item => item.code === code) || null;
} 